src/System.Security.Cryptography.Xml/tests/Samples/EncryptingDecryptingAsymmetric.cs

```diff
@@ -0,0 +1,102 @@
+    // https://msdn.microsoft.com/en-us/library/ms229746(v=vs.110).aspx
+    public class EncryptingAndDecryptingAsymmetric
+    {
+        const string ExampleXmlRootElement = "example";
+        const string ExampleXml = @"<?xml version=""1.0""?>
+<example>
+<test>some text node</test>
+</example>";
+
+        private static XmlDocument LoadXmlFromString(string xml)
+        {
+            var doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+            return doc;
+        }
+
+        private static void Encrypt(XmlDocument doc, string elementName, string encryptionElementID, RSA rsaKey, string keyName)
+        {
+            var elementToEncrypt = (XmlElement)doc.GetElementsByTagName(elementName)[0];
+
+            using (var sessionKey = new RijndaelManaged())
+            {
+                sessionKey.KeySize = 256;
+
+                // Encrypt the session key and add it to an EncryptedKey element.
+                var encryptedKey = new EncryptedKey()
+                {
+                    CipherData = new CipherData(EncryptedXml.EncryptKey(sessionKey.Key, rsaKey, false)),
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncRSA15Url)
+                };
+
+                // Specify which EncryptedData
+                // uses this key. An XML document can have
+                // multiple EncryptedData elements that use
+                // different keys.
+                encryptedKey.AddReference(new DataReference()
+                {
+                    Uri = "#" + encryptionElementID
+                });
+
+                var encryptedData = new EncryptedData()
+                {
+                    Type = EncryptedXml.XmlEncElementUrl,
+                    Id = encryptionElementID,
+
+                    // Create an EncryptionMethod element so that the
+                    // receiver knows which algorithm to use for decryption.
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncAES256Url)
+                };
+
+                encryptedData.KeyInfo.AddClause(new KeyInfoEncryptedKey(encryptedKey));
+                encryptedKey.KeyInfo.AddClause(new KeyInfoName()
+                {
+                    Value = keyName
+                });
+
+                var encryptedXml = new EncryptedXml();
+                encryptedData.CipherData.CipherValue = encryptedXml.EncryptData(elementToEncrypt, sessionKey, false);
+
+                EncryptedXml.ReplaceElement(elementToEncrypt, encryptedData, false);
+            }
+        }
+
+        public static void Decrypt(XmlDocument doc, RSA rsaKey, string keyName)
+        {
+            var encrypted = new EncryptedXml(doc);
+            encrypted.AddKeyNameMapping(keyName, rsaKey);
+            encrypted.DecryptDocument();
+        }
+
+        [Fact]
+        public void AsymmetricEncryptionRoundtrip()
+        {
+            using (var key = RSA.Create())
```

Comment:
@morganbr i saw in some examples we use Clear but the documentation claims that Dispose should clean all the native resources. I think whatever is happening if there needs to be any fix I'd rather fix RSA class